src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMATE = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String DASH = " - ";
+    private static final int INDEX_ONE = 1;
+    private static final int INDEX_THREE = 3;
+    private static final int INDEX_TWO = 2;
+    private static final int INDEX_ZERO = 0;
+    private static final String REPORT_TITLE = "Report for period ";
+    private static final String SPLIT_REGEXP = " ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate from = LocalDate.parse(dateFrom, DATE_FORMATE);
+        LocalDate to = LocalDate.parse(dateTo, DATE_FORMATE);
+        StringBuilder resultStrBuilder = new StringBuilder();
+        resultStrBuilder.append(REPORT_TITLE)
+                .append(dateFrom)
+                .append(DASH)
+                .append(dateTo)
+                .append(System.lineSeparator());
+
+        generateData(names, data, from, to, resultStrBuilder);
+
+        return resultStrBuilder.toString().trim();
+    }
+
+    private void generateData(String[] names,
```

Comment:
Are you sure that this method generates something?

Reply:
Sorry, my bad, fixed.